api/tls_context.proto

```diff
@@ -46,19 +46,24 @@ message CertificateValidationContext {
   // presented it will not be verified.
   DataSource ca_cert = 1;
 
-  // If specified, Envoy will verify (pin) the hash of the presented
-  // certificate.
+  // If specified, Envoy will verify (pin) hex-encoded SHA-256 hash of
+  // the presented certificate.
   repeated string verify_certificate_hash = 2;
 
+  // If specified, Envoy will verify (pin) base64-encoded SHA-256 hash of
+  // the Subject Public Key Information (SPKI) of the presented certificate.
+  // This is the same format as used in HTTP Public Key Pinning.
+  repeated string verify_spki_sha256 = 3;
```

Comment:
I think we should allow both schemes at the same time, as well as `verify_subject_alt_names`, and allow connection if it matches any of them.